File: crates/atlas-cli/tests/fixtures/go/simple.go
package main

import "fmt"

// Server is a simple HTTP server.
type Server struct {
	Port int
}

// Start begins listening on the configured port.
func (s *Server) Start() {
	fmt.Printf("listening on :%d\n", s.Port)
}

func main() {
	srv := Server{Port: 8080}
	srv.Start()
}

File: crates/atlas-cli/tests/fixtures/javascript/simple.js
// Golden test fixture: JavaScript simple
// Covers: class/method/variable definitions, call/field access/reference, import/export, scope nesting

import { Router } from "express";

export class Server {
  #port;

  constructor(port) {
    this.#port = port;
  }

  start() {
    const router = new Router();
    router.listen(this.#port);
    console.log("started");
  }
}

function createServer(port) {
  return new Server(port);
}

export { createServer };
